lib/Doctrine/CouchDB/HTTP/MultipartParserAndSender.php

```diff
@@ -0,0 +1,413 @@
+                //  1) Array of json string documents that don't have attachments. These should be posted using the bulk
+                //     api.
+                //  2) Responses of posting docs with attachments.
+                $body = $this->parseAndSend($targetPath);
+            }
+            return $body;
+        }
+    }
+
+
+    /**
+     * Read and return next line from the connection pointer.
+     * $maxLength parameter can be used to set the maximum length
+     * to be read.
+     *
+     * @param int $maxLength
+     * @return string
+     */
+    protected function getNextLineFromSourceConnection($maxLength = null)
+    {
+        if ($maxLength !== null) {
+            return fgets($this->sourceConnection, $maxLength);
+        } else {
+            return fgets($this->sourceConnection);
+        }
+    }
+
+    /**
+     * Parses multipart data. Returns an array having:
+     * 1) Array of json docs(string) that don't have attachments. These
+     * should be posted using the bulk api. and
+     * 2) Responses of posting docs with attachments.
+     *
+     * @param $targetPath
+     * @return array
+     * @throws \Exception
+     * @throws \HTTPException
+     */
+    protected function parseAndSend($targetPath)
+    {
+        // Main response boundary of the multipart stream.
+        $mainBoundary = trim($this->getNextLineFromSourceConnection());
+
+        // Docs that don't have attachment.
+        // These should be posted using bulk upload.
+        $docStack = array();
+
+        // Responses from posting docs that have attachments.
+        $responses = array();
+
+        while (!feof($this->sourceConnection)) {
+
+            $line = ltrim($this->getNextLineFromSourceConnection());
+            if ($line == '') {
+                continue;
+
+            } elseif (strpos($line, "Content-Type") !== false) {
+
+
+                list($header, $value) = explode(":", $line);
+                $header = trim($header);
+                $value = trim($value);
+                $boundary = '';
+
+                if (strpos($value, ";") !== false) {
+                    list($type, $info) = explode(";", $value, 2);
+                    $info = trim($info);
+
+                    // Get the boundary for the current doc.
+                    if (strpos($info, "boundary") !== false) {
+                        $boundary = $info;
+
+                    } elseif (strpos($info, "error") !== false) {
+
+                        // Missing revs at the source.
+                        // Continue till the end of this document.
+                        while (strpos($this->getNextLineFromSourceConnection(), $mainBoundary) === false) ;
+
+                    } else {
+
+                        throw new \Exception("Unknown parameter with Content-Type.");
+                    }
+
+                }
+                // Doc with attachments.
+                if (strpos($value, "multipart/related") !==false) {
```

Comment:
Needs proper spacing between the `!==` comparison. And maybe single quotation marks.